src/EditorFeatures/CSharp/ConvertNamespace/ConvertNamespaceCommandHandler.cs

```diff
@@ -0,0 +1,175 @@
+        }
+
+        /// <summary>
+        /// Returns the updated file contents if semicolon is typed after a block-scoped namespace name that can be
+        /// converted.
+        /// </summary>
+        private CompilationUnitSyntax? ConvertNamespace(
+            TypeCharCommandArgs args,
+            CommandExecutionContext executionContext)
+        {
+            if (args.TypedChar != ';' || !args.TextView.Selection.IsEmpty)
+                return null;
+
+            if (!_globalOptions.GetOption(FeatureOnOffOptions.AutomaticallyCompleteStatementOnSemicolon))
+                return null;
+
+            var subjectBuffer = args.SubjectBuffer;
+            var caretOpt = args.TextView.GetCaretPoint(subjectBuffer);
+            if (!caretOpt.HasValue)
+                return null;
+
+            var caret = caretOpt.Value.Position;
+            var document = subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null)
+                return null;
+
+            var cancellationToken = executionContext.OperationContext.UserCancellationToken;
+            var root = (CompilationUnitSyntax)document.GetRequiredSyntaxRootSynchronously(cancellationToken);
+
+            // User has to be *after* an identifier token.
+            var token = root.FindToken(caret);
+            if (token.Kind() != SyntaxKind.IdentifierToken)
+                return null;
+
+            if (caret < token.Span.End ||
+                caret >= token.FullSpan.End)
+            {
+                return null;
+            }
+
+            var namespaceDecl = token.GetRequiredParent().GetAncestor<NamespaceDeclarationSyntax>();
+            if (namespaceDecl == null)
+                return null;
+
+            // That identifier token has to be the last part of a namespace name.
+            if (namespaceDecl.Name.GetLastToken() != token)
+                return null;
+
+            // Pass in our special options, and C#10 so that if we can convert this to file-scoped, we will.
+            if (!ConvertNamespaceAnalysis.CanOfferUseFileScoped(s_optionSet, root, namespaceDecl, forAnalyzer: true, LanguageVersion.CSharp10))
+                return null;
+
+            var fileScopedNamespace = (FileScopedNamespaceDeclarationSyntax)ConvertNamespaceTransform.Convert(namespaceDecl);
+
+            // Place an annotation on the semicolon so that we can find it post-formatting to place the caret.
+            fileScopedNamespace = fileScopedNamespace.WithSemicolonToken(
+                fileScopedNamespace.SemicolonToken.WithAdditionalAnnotations(s_annotation));
+
+            var convertedRoot = root.ReplaceNode(namespaceDecl, fileScopedNamespace);
+            var formattedRoot = (CompilationUnitSyntax)Formatter.Format(
+                convertedRoot, Formatter.Annotation,
+                document.Project.Solution.Workspace,
+                options: null, rules: null, cancellationToken);
```

Comment:
❗ This assumes the file is formatted according to the formatter, which is not a safe assumption. This feature is only allowed to remove a uniform number of leading spaces from each line in the namespace block, and is not allowed to force potentially incorrect formatting throughout the file.

Reply:
I saw this tweet that indicated the codefix for converting to file scoped namespaces forced a format on users https://twitter.com/rickbrewPDN/status/1463361397677379590

Reply:
Yah, this has come up on Discord a couple of times too, and ASP.NET ran into it too: https://github.com/dotnet/aspnetcore/pull/38681#discussion_r758007177